### PR TITLE
pauloue's ItemStack example

### PR DESCRIPTION
**Pull request by FreeLikeGNU**
This adds pauloue's ItemStack example from PR  #8847 to the api doc.  It helped me understand how this arcane system operates.

### CODE REVIEW

**Comment by Wuzzy2** [comment]
This is not particulary good. It requires knowledge of internal MTG node names, and knowledge which node names actually exist, and which ones are fake.

It's better to work with fictional node names (like `example:something`) and indicate how these items are defined (e.g. item 1 has description, item 2 does not, item 3 is fake and doesnt exist at all).

**Comment by SmallJoker** [comment]
The Lua API provides an overall explanation of the features. Specific examples should be better written down in the wiki (dev.minetest.net).

**Comment by FreeLikeGNU** [comment]
With modders being actively discouraged to refer to the wiki, I'm curious why that has been suggested, @SmallJoker? 

As for @Wuzzy2's  and @SmallJoker concern, there are many "specific examples" of code in the lua_api.txt  here's one: 
LINE 451: `  [combine:16x32:0,0=default_cobble.png:0,16=default_wood.png`

Many other such examples with specific content appear throughout the document.

I think it's important for both fictional and easily testable examples using content available in minimal default mod exist, however.  If this needs to be rewritten to reflect such available content, I'm happy to revise it.   
Thank you for your feedback.

**Comment by appgurueu** [comment]
I'd just document the additional behavior demonstrated in the examples:

* `get_description()` returns `Unknown Item` for unknown items (this is bad IMO and should not be relied upon)
* Defined items without description return the itemname as description (bad as well IMO)
* The `description` meta field can be used to store a different description

File: doc/lua_api.txt [hunk 1]
```diff
@@ -1974,6 +1974,14 @@ Example:
     meta:set_string("key", "value")
     print(dump(meta:to_table()))
 
+Example manipulations of "description" and expected output behaviors:
+
+    print(ItemStack("default:pick_steel"):get_description()) --> Steel Pickaxe
+    print(ItemStack("foobar"):get_description()) --> Unknown Item
+    local stack = ItemStack("default:stone")
```
Comment (rubenwardy) [inline doc/lua_api.txt:1981]:
```suggestion

    local stack = ItemStack("default:stone")
```

File: doc/lua_api.txt [hunk 2]
```diff
@@ -1974,6 +1974,14 @@ Example:
...
+    stack:get_meta():set_string("description", "Custom description")
+    print(stack:get_description()) --> Custom description
+    print(ItemStack("beds:bed_top"):get_description()) --> beds:bed_top
```
Comment (rubenwardy) [inline doc/lua_api.txt:1984]:
```suggestion

    print(ItemStack("mod:item_with_no_desc"):get_description()) --> mod:item_with_no_desc
```

**Review by rubenwardy — APPROVED** [review]

**Review by sfan5 — APPROVED** [review]

**Pull request merged**